index.js

```diff
@@ -8,62 +8,36 @@ var platforms = {
 };
 
 
-exports.verifyPayment = function (platform, payment, cb) {
-	function syncError(error) {
-		process.nextTick(function () {
-			cb(error);
-		});
-	}
-
-	if (!payment) {
-		return syncError(new Error('No payment given'));
-	}
-
-	var engine = platforms[platform];
-
-	if (!engine) {
-		return syncError(new Error('Platform ' + platform + ' not recognized'));
-	}
-
-	engine.verifyPayment(payment, function (error, result) {
-		if (error) {
-			return cb(error);
+function makeMethod(methodName) {
+	exports[methodName] = function (platform, payment, cb) {
+		function asyncError(message) {
+			var error = new Error(message);
+			// this will cause the stack trace to start at the place where asyncError
+			// is called
+			error.captureStackTrace(error, asyncError);
+			process.nextTick(cb, error);
 		}
 
-		result.platform = platform;
-
-		cb(null, result);
-	});
-};
-
-
-exports.cancelSubscription = function (platform, payment, cb) {
-	function syncError(error) {
-		process.nextTick(function () {
-			cb(error);
-		});
-	}
+		if (!payment) {
+			return asyncError('No payment given');
+		}
 
-	if (!payment) {
-		return syncError(new Error('No payment given'));
-	}
+		var engine = platforms[platform];
 
-	var engine = platforms[platform];
+		if (!engine) {
+			return asyncError('Platform ' + platform + ' not recognized');
+		}
 
-	if (!engine) {
-		return syncError(new Error('Platform ' + platform + ' not recognized'));
-	}
+		if (!engine[methodName]) {
+			return asyncError(
+				'Platform ' + platform + ' does not have a ' + methodName + ' method'
```

Comment:
I'd have used string interpolation via [ES6 Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) but wasn't sure if that would be appropriate since I don't see any ES6 futures being used in the codebase (although the linter seems to be asking for `let`, `const` etc.).

Reply:
Tbh, I'm not sure from which Node version that has been supported. But in general, I think we can move things over into ES6/7 land. ESLint is already configured to allow pretty much anything.

Reply:
for ESX compatibility see https://node.green/

Template literals [have been available natively since node 4](https://node.green/#ES2015-syntax-template-literals)

Personally I'd only use ES6 and beyond if the package.json specified the lowest compatible node version and CI was configured to test the code with that version. Another option is to go full on and use Babel.

Reply:
Babel is awesome, but for non-browser projects I prefer not to use a transpiler. There's no need to support Node 4 anymore, so let's stick to Node 6, 8, 10, based on the official [schedule](https://github.com/nodejs/Release/blob/master/README.md).